### PR TITLE
Certificate

### PR DESCRIPTION
**Pull request by tommygonzaleza**

### CODE REVIEW

**Comment by tommygonzaleza** [comment]
https://github.com/breatheco-de/breatheco-de/issues/5869

File: breathecode/certificate/receivers.py [hunk 1]
```diff
@@ -14,3 +16,10 @@ def post_save_user_specialty(sender, instance: UserSpecialty, **kwargs):
 
     elif instance._hash_was_updated and instance.status == 'PERSISTED' and not instance.preview_url:
         tasks.take_screenshot.delay(instance.id)
+
+
+@receiver(student_edu_status_updated, sender=CohortUser)
+def generate_certificate(sender, instance: CohortUser, **kwargs):
+    print(instance)
+    if instance.educational_status == 'GRADUATED':
```
Comment (jefer94) [inline breathecode/certificate/receivers.py:24]: The certificate just must be autogenerated if the student coming from a saas cohort or academy cohort, if cohort.is_available_as_saas in [False] it does not take in consideration the value in the academy

**Review by jefer94 — CHANGES_REQUESTED** [review]

File: breathecode/marketing/migrations/0074_alter_formentry_attribution_id.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
+# Generated by Django 3.2.18 on 2023-08-31 19:57
```
Comment (jefer94) [inline breathecode/marketing/migrations/0074_alter_formentry_attribution_id.py:1]: The migrations didn't run

File: breathecode/certificate/actions.py [hunk 3]
```diff
@@ -135,13 +135,14 @@ def generate_certificate(user, cohort=None, layout=None):
...
-        if cohort.current_day != cohort.syllabus_version.syllabus.duration_in_days:
+        if (not cohort.never_ends
+            ) and cohort.current_day != cohort.syllabus_version.syllabus.duration_in_days:
             raise ValidationException(
                 'Cohort current day should be '
                 f'{cohort.syllabus_version.syllabus.duration_in_days}',
                 slug='cohort-not-finished')
 
-        if cohort.stage != 'ENDED':
+        if (not cohort.never_ends) and cohort.stage != 'ENDED':
```
Comment (jefer94) [inline breathecode/certificate/actions.py:145]: Changes in the logic of certificate require tests

**Review by jefer94 — CHANGES_REQUESTED** [review]

**Pull request merged**